src/algebra/group_ring_action.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+Group action on rings.
+-/
+
+import group_theory.group_action data.equiv.ring
+
+universes u v
+
+variables (α : Type u) (β : Type v)
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+
+/-- Typeclass for multiplicative actions by monoids on monoids. -/
+class mul_mul_action [monoid α] [monoid β] extends mul_action α β :=
+(smul_one : ∀ (g : α), (g • 1 : β) = 1)
+(smul_mul : ∀ (g : α) (x y : β), g • (x * y) = (g • x) * (g • y))
+
+/-- Typeclass for multiplicative actions by monoids on additive monoids. -/
+class mul_add_action [monoid α] [add_monoid β] extends mul_action α β :=
```

Comment:
This already exists under the name `distrib_mul_action`

Reply:
Should I keep the `to_additive`?

Reply:
I guess we need a refactor :sob:

Reply:
BTW, could you please use descriptive type variable names? E.g., `M` for a monoid and `R` for a ring. We should add this to `naming.md`. And there should be an instance for `algebra` being a `mul_semiring_action`.